core/CMakeLists.txt

```diff
@@ -43,21 +44,40 @@ if (APPLE)
             src/main/c/osx/accept.c
     )
 elseif (UNIX)
-    include_directories($ENV{JAVA_HOME}/include/linux/)
-    message(${CMAKE_SYSTEM_PROCESSOR})
-    if (CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64")
-        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/armlinux)
-        set(AARCH64 1)
-    else ()
-        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/linux)
-    endif ()
-    set(
+    EXEC_PROGRAM(uname OUTPUT_VARIABLE SYSTEM_NAME)
+    SET(SYSTEM_NAME "${SYSTEM_NAME}" CACHE INTERNAL "")
```

Comment:
This is a really old school to check system name.  There is a decent (and up to date) guide on how to do this that can be found here:

https://gitlab.kitware.com/cmake/community/-/wikis/doc/tutorials/How-To-Write-Platform-Checks

If above works it would be much better to avoid accidents such as running `uname` on windows. I know it may not run on windows, but still, external command exec isn't great.

Reply:
I was looking for that... to avoid the EXEC

> build also failed. CircleCI runs on Linux.